client/state/comments/reducer.js

```diff
@@ -191,6 +192,38 @@ export const totalCommentsCount = createReducer(
 );
 
 /**
+ * Stores a site's comments tree.
+ * @param {Object} state Redux state (`comments.trees`)
+ * @param {Object} action Redux action
+ * @returns {Object} New Redux state.
+ */
+export const trees = ( state = {}, { siteId, status, tree, type } ) => COMMENTS_TREE_RECEIVE === type
+	? {
+		...state,
+		[ siteId ]: {
+			...state[ siteId ],
+			[ status ]: tree,
+		}
+	}
+	: state;
+
```

Comment:
check out `state/utils#keyedReducer` for this kind of reducer to remove the awkward state spreads

```js
export const treeItem = ( state, { tree, type } ) =>
	COMMENTS_TREE_ADD === type
		? insertTree( state, tree )
		: state;

export const commentsTree = keyedReducer( 'siteId', treeItem );
```